src/main/java/org/elasticsearch/script/ScriptMode.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import org.elasticsearch.ElasticsearchIllegalArgumentException;
+
+import java.util.Locale;
+
+/**
+ * Mode for a specific script, used for script settings.
+ * Defines whether a certain script or catefory of scripts can be executed or not, or whether it can
+ * only be executed by a sandboxed scripting language.
+ */
+enum ScriptMode {
+    ENABLE,
+    DISABLE,
+    SANDBOX;
+
+    static ScriptMode parse(String input) {
+        input = input.toLowerCase(Locale.ROOT);
+        switch(input) {
```

Comment:
we can but we usually tend to be "understanding" with user input. I don't think this is a big problem, I actually did it because the suggested syntax ("enable" and "disable") differs from what we do in other places. Also whenever we accept a boolean we accept `on`, `off`, `true` and `false` if I remember correctly